bluzelled/src/NodeUtilities.cpp

```diff
@@ -53,9 +52,47 @@ void remove_dead_nodes(Nodes *nodes)
     //    std::cout << "dead nodes: " << rnodes->size() << std::endl;
 }
 
-
 void reaper(Nodes *nodes)
 {
     join_dead_tasks(nodes);
     remove_dead_nodes(nodes);
 }
+
+void kill_nodes(const Nodes &nodes)
+{
+    for (auto node : nodes)
+        {
+        node->kill();
+        }
+}
+
+void join_node_threads(const Nodes &nodes)
+{
+    for (auto node : nodes)
+        {
+        node->join();
+        }
+}
+
+void kill_and_join_all_nodes(Nodes &nodes)
+{
+    kill_nodes(nodes);
+    join_node_threads(nodes);
+}
+
+
+Node *create_node(std::shared_ptr<Listener> l)
+{
+    return new Node(l);
+}
+
+
+Nodes create_nodes(int number_of_nodes, std::shared_ptr<Listener> l)
+{
+    Nodes nodes;
+    for (int i = 0; i < number_of_nodes; ++i)
+        {
+        nodes.emplace_back(create_node(l));
```

Comment:
@rich AFAIK emplace_bake() method takes ctor parameters so it can be used as emplace_back(l). Create node method is redundant.